core/geometry/src/polyface/IndexedPolyfaceWalker.ts

```diff
@@ -0,0 +1,350 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module Polyface
+ */
+
+import { IndexedEdgeMatcher, SortableEdgeCluster } from "./IndexedEdgeMatcher";
+import { IndexedPolyfaceVisitor } from "./IndexedPolyfaceVisitor";
+import { IndexedPolyface } from "./Polyface";
+
+/**
+ * The `IndexedPolyfaceWalker` class supports navigation around facets, across edges, and around vertices in an
+ * [[IndexedPolyface]].
+ * * Compare to the [[IndexedPolyfaceVisitor]] class, which supports the iteration of facets in an `IndexedPolyface`.
+ * * A one-time call to [[buildEdgeMateIndices]] creates and populates the `data.edgeMateIndex` array on the input.
+ *   * This array essentially completes the topology of the `IndexedPolyface` by storing facet adjacency.
```

Comment:
what do you mean by "completes"? Was the topoly "incomplete" before this PR?

Reply:
Yes, it was incomplete. `PolyfaceData` did not store facet adjacency.